src/board/board_test.lua

```diff
@@ -0,0 +1,353 @@
+        for i = 0, 9, 1 do
+            table.insert(board.boardTetrominoSquares, Square:new(Point:new(i, 0), colors.ASH))
+            table.insert(board.boardTetrominoSquares, Square:new(Point:new(i, 2), colors.ASH))
+        end
+        table.insert(board.boardTetrominoSquares, Square:new(Point:new(0, 1), colors.ASH))
+        table.insert(board.boardTetrominoSquares, Square:new(Point:new(1, 1), colors.ASH))
+        table.insert(board.boardTetrominoSquares, Square:new(Point:new(2, 1), colors.ASH))
+        table.insert(board.boardTetrominoSquares, Square:new(Point:new(0, 3), colors.ASH))
+        table.insert(board.boardTetrominoSquares, Square:new(Point:new(1, 3), colors.ASH))
+
+        board:updateMatrices()
+        filledIndices = board.boardTetrominosMatrix:getFilledIndices()
+        board:clearLines(filledIndices)
+        board:dropLines(filledIndices)
+        board:updateMatrices()
+
+        for i = 0, 2, 1 do
+            assert.is_true(board.boardTetrominosMatrix:isFilled(i, 0))
+        end
+        for i = 3, 9, 1 do
+            assert.is_false(board.boardTetrominosMatrix:isFilled(i, 0))
+        end
+
+        for i = 0, 1, 1 do
+            assert.is_true(board.boardTetrominosMatrix:isFilled(i, 1))
+        end
+        for i = 2, 9, 1 do
+            assert.is_false(board.boardTetrominosMatrix:isFilled(i, 1))
+        end
+    end)
+
+    it("ObstacleBelow", function()
+        assert.is_false(board:obstacleBelow(board.currentTetromino.squares))
+
+        -- testing to see if at the bottom of board
+        board.currentTetromino:offset(0, -19)
+        assert.is_true(board:obstacleBelow(board.currentTetromino.squares))
+
+        -- add squares in the way and test if there is an obstacle below
+        board.currentTetromino = board.nextTetromino
+        table.insert(board.boardTetrominoSquares, Square:new(Point:new(3, 0), colors.ASH))
+        table.insert(board.boardTetrominoSquares, Square:new(Point:new(4, 0), colors.ASH))
+        table.insert(board.boardTetrominoSquares, Square:new(Point:new(5, 0), colors.ASH))
+        table.insert(board.boardTetrominoSquares, Square:new(Point:new(4, 1), colors.ASH))
+
+        board:updateMatrices()
+        board.currentTetromino:offset(0, -17)
+        board:obstacleBelow(board.currentTetromino.squares)
+    end)
+
+    it("HandleGravity", function()
+        -- testing gravity timer
+        assert.are_equal(0, board.gravityFrameCounter.elapsedFrames)
+        board.gravityFrameCounter.maxFrames = 5
+        currentTetrominoID = board.currentTetromino.id
+        board:handleGravity(10)
+
+        -- gravity timer should have reset and moved down one
+        movedDownTetromino = Tetromino:new(currentTetrominoID,
+                properties.SPAWN[currentTetrominoID], colors.ASH)
+        movedDownTetromino:offset(0, -1)
+        assert.are_equal(0, board.gravityFrameCounter.elapsedFrames)
+        assert.are.same(movedDownTetromino.origin, board.currentTetromino.origin)
```

Comment:
This just checks that currentTetromino moved down right?
I would just do something like 
```suggestion
        assert.are.equal(18, board.currentTetromino.origin.y)
```
and instantiate it at height 19 above